python/pyspark/sql/dataframe.py

```diff
@@ -1721,7 +1721,14 @@ def toPandas(self):
         1    5    Bob
         """
         import pandas as pd
-        return pd.DataFrame.from_records(self.collect(), columns=self.columns)
+
+        dtype = {}
+        for field in self.schema:
+            pandas_type = _to_corrected_pandas_type(field.dataType)
+            if (pandas_type):
+                dtype[field.name] = pandas_type
+
+        return pd.DataFrame.from_records(self.collect(), columns=self.columns).astype(dtype)
```

Comment:
The param `copy` of `astype` is true by default. Seems to me we don't need copying the data?